### PR TITLE
Helm metabase

### PR DESCRIPTION
**Pull request by thegentlemanphysicist**
This helm chart allows a user to deploy to the gold tools production namespace, please confirm that the network policies are set up correctly.

### CODE REVIEW

File: helm/metabase/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+SHELL := /usr/bin/env bash
+
+
+# This installs metabase in the gold production tools namespace
+.PHONY: install
+install:
+	helm install metabase . -n eb75ad-tools \
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply helm/metabase/Makefile]: resolved

File: helm/metabase/Makefile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	--set postgresql_password="$(shell bash -c 'echo $$RANDOM | md5sum | head -c 32')"
+
+# Upgrade will change an existing metabase deployment. The secret gets regenerated each time this is run
+.PHONY: upgrade
+upgrade:
+	helm upgrade --install metabase . \
+	-n eb75ad-tools \
+	-f values.yaml \
+	-f "values-eb75ad-tools.yaml" \
+	--set postgresql_password="$(shell bash -c 'echo $$RANDOM | md5sum | head -c 32')"
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply helm/metabase/Makefile:19]: Renamed

File: helm/metabase/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply helm/metabase/Makefile]: It is, but I have generalized the make file

File: helm/metabase/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+```
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: sso-dev-production-gold-metabase-access
+  namespace: eb75ad-dev
+spec:
+  podSelector:
+    matchLabels:
+      app: sso-patroni
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply helm/metabase/README.md]: resolved

File: helm/metabase/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+### Set Network policies
+
+Metabase needs to access other namespaces.  To allow access we must create a network policy in the namespace holding the target database.  For the sso-patroni databases, this network policy is:
+
+```
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: sso-dev-production-gold-metabase-access
+  namespace: eb75ad-dev
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply helm/metabase/README.md]: updated documentation to be more general

File: helm/metabase/templates/imagestream.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply helm/metabase/templates/imagestream.yaml]: Removed the image stream and service account

File: helm/metabase/templates/deploymentconfig.yaml [hunk 6]
```diff
@@ -0,0 +1,186 @@
+apiVersion: apps.openshift.io/v1
+kind: DeploymentConfig
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply helm/metabase/templates/deploymentconfig.yaml:2]: I'm going to leave this as is for now.  If we end up needing to generalize this to all kubernetes clusters, I'm happy do do it when it becomes something our team needs.

File: helm/metabase/values.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
+
+## These are the values that were needed by the Metabase helm templates generated by template2helm
```
Comment (junmin-bcgov) [inline helm/metabase/values.yaml:37]: I think I forgot to add comments on this. can we update this block as:
```yaml
image:
  repository: docker-remote.artifacts.developer.gov.bc.ca/metabase/metabase
  tag: latest
  pullPolicy: Always

postgresql:
  database: metabase
  password: # Generated in the make file
  user: metabase
  version: "9.6"
  volumeCapacity: 1Gi
```

**Review by junmin-bcgov — COMMENTED** [review]
I had a one comment and wonder if it won't work without the `ImageStream`?

**Review by junmin-bcgov — APPROVED** [review]

**Pull request merged**